public_html/wp-content/plugins/wordcamp-attendee-survey/wordcamp-attendee-survey.php

```diff
@@ -0,0 +1,188 @@
+<?php
+/**
+ * Plugin Name:     WordCamp Attendee Survey
+ * Plugin URI:      https://wordcamp.org
+ * Description:     Send survey to WordCamp attendees.
+ * Author:          WordCamp.org
+ * Author URI:      https://wordcamp.org
+ * Version:         1
+ *
+ * @package         WordCamp\AttendeeSurvey
+ */
+
+namespace WordCamp\AttendeeSurvey;
+
+use function WordCamp\AttendeeSurvey\Email\{add_email, delete_email};
+use function WordCamp\AttendeeSurvey\Page\{add_page,delete_page};
+
+defined( 'WPINC' ) || die();
+
+/**
+ * Local dependencies.
+ */
+require_once get_includes_path() . 'email.php';
+require_once get_includes_path() . 'page.php';
+
+/**
+ * Plugin activation and deactivation hooks.
+ */
+register_activation_hook( __FILE__, __NAMESPACE__ . '\activate' );
+register_deactivation_hook( __FILE__, __NAMESPACE__ . '\deactivate' );
```

Comment:
IIRC this only runs once, when the plugin is manually activated. It won't run when new sites get created after the plugin has been activated.

It may be better to put all of those things in `wordcamp-new-site.php` (or keep the code in this plugin, but hook into one of the functions in that file).

Reply:
I decided to use `init` and return early if an option with the feature id exists. I chose to do so because it makes this code easy to delete and separated from other plugins in this repo. There is a down side of course. If the option ever gets deleted, it will setup again and could lead to duplicate pages/emails. I think it's worth it although I could be easily swayed the other way. I also think performance wise its probably inconsequential.

Reply:
Ah, I see, that works 👍🏻 

It feels a little unconventional to me (for WordCamp code at least), and the code could be a good amount leaner without it IMO. That's just personal preference though, I don't have an objection to it.